compiler/test/dotty/tools/backend/jvm/DottyBytecodeTests.scala

```diff
@@ -695,4 +695,50 @@ class TestBCode extends DottyBytecodeTest {
         "`test` was not properly generated\n" + diffInstructions(instructions, expected))
     }
   }
+
+  @Test def i5924b = {
+    val source =
+      """|import scala.annotation.static
+         |trait Base
+         |
+         |object Base {
+         |  @static val x = 10
+         |  @static final val y = 10
+         |  @static def f: Int = 30
+         |}
+      """.stripMargin
+
+    checkBCode(source) { dir =>
+      val clsIn   = dir.lookupName("Base.class", directory = false).input
+      val clsNode = loadClassNode(clsIn)
+      getMethod(clsNode, "f")
```

Comment:
Would be good to check that the correct flags are set, you can call `.access` on a MethodNode or FieldNode to get its flags: https://static.javadoc.io/org.ow2.asm/asm/6.0/org/objectweb/asm/tree/MethodNode.html#access, and then check if ACC_STATIC and ACC_FINAL are enabled.